assets/js/event-filtering.js

```diff
@@ -0,0 +1,67 @@
+$(function() {
+    var eventsElements = $(".event-tags span");
+    var eventFilterParent = $("#eventFilter");
+    var tags = [];
+
+    for (var i = 0; i < eventsElements.length; i++) {
+        var elem = eventsElements[i];
+        var text = $(elem).text();
+        if (tags.indexOf(text) === -1) {
+            tags.push(text);
+            eventFilterParent.append(
+                '<input type="checkbox" class="mx-2" value="' + text + '">' + text + '</p>'
+            );
+        }
+    }
+
+    $("input[type='checkbox']").on("click", function() {
+        var inputs = $("input[type='checkbox']");
+        var chosenInputs = [];
+
+        for (var z = 0; z < inputs.length; z++) {
+            var input = inputs[z];
+            var isChecked = $(input).prop("checked");
+            if (isChecked) {
+                chosenInputs.push($(input).val().toLowerCase());
+            }
+        }
+
+        filterEventList(chosenInputs);
+    });
+
+    /**
+     *
+     * @param {*[]} arr1 An array
+     * @param {*[]} arr2 An array
+     */
+    function checkForIntersection(arr1, arr2) {
+        for (var i = 0; i < arr1.length; i++) {
+            var val = arr1[i];
+            var hasIntersection = arr2.indexOf(val) > -1;
```

Comment:
Nit: IMHO just writing this as `if (arr2.indexOf(val) !== -1)` seems a little cleaner.

But `indexOf` will only work using referential equality, right?

```javascript
var arr1 = [ { foo: "alpha" } ];
var arr2 = [ { foo: "alpha" } ];

// Returns false, since the objects are different references.
checkForIntersection(arr1, arr2)
// Returns true, since the objects would have the same reference.
checkForIntersection(arr1, arr1)
```

Right? Probably worth clarifying in the comment above, IMHO.

Reply:
Yup, you're right. Clarified in the comment